data/data-pipeline/data_pipeline/etl/sources/cdc_life_expectancy/etl.py

```diff
@@ -23,32 +35,122 @@ def __init__(self):
             self.LIFE_EXPECTANCY_FIELD_NAME,
         ]
 
+        # Set some constants that will be helpful for debugging the source data later.
+        self.STATE_FIPS_CODES = get_state_fips_codes(self.DATA_PATH)
+
+        self.EXPECTED_STATES_SET = (
+            set(self.STATE_FIPS_CODES)
+            # We don't expect LEEP to have data for island areas or Puerto Rico.
+            - set(TILES_ISLAND_AREA_FIPS_CODES)
+            - set(TILES_PUERTO_RICO_FIPS_CODE)
+        )
+
+        # These states are currently missing from LEEP's whole USA file.
+        self.EXPECTED_MISSING_STATES = ["23", "55"]
+
         self.raw_df: pd.DataFrame
         self.output_df: pd.DataFrame
 
     def extract(self) -> None:
         logger.info("Starting data download.")
 
-        download_file_name = (
+        all_usa_download_file_name = (
             self.get_tmp_path() / "cdc_life_expectancy" / "usa.csv"
         )
         download_file_from_url(
-            file_url=self.FILE_URL,
-            download_file_name=download_file_name,
+            file_url=self.USA_FILE_URL,
+            download_file_name=all_usa_download_file_name,
+            verify=True,
+        )
+
+        pandas_read_csv_dtype_settings = {
+            # The following need to remain as strings for all of their digits, not get converted to numbers.
+            self.TRACT_INPUT_COLUMN_NAME: "string",
+            self.STATE_INPUT_COLUMN_NAME: "string",
+        }
+
+        all_usa_raw_df = pd.read_csv(
+            filepath_or_buffer=all_usa_download_file_name,
+            dtype=pandas_read_csv_dtype_settings,
+            low_memory=False,
+        )
+
+        # Check which states are missing
+        states_in_life_expectancy_usa_file = all_usa_raw_df[
+            self.STATE_INPUT_COLUMN_NAME
+        ].unique()
+
+        # Find which states are missing from the expected set.
+        states_missing = sorted(
```

Comment:
this is a confusing way to do this and relies on sorting -- when you could just check the set difference and the set union and ensure it's the same as the set of states.

Reply:
oh also "nit"

Reply:
good point!! fixing now